app/assets/javascripts/settings/ldap-conn-test.js

```diff
@@ -0,0 +1,78 @@
+$(function () {
+  function NoneEmpty (arr) {
+    for (var i = 0; i < arr.length; i++) {
+      if (arr[i] === '') return false
+    }
+    return true
+  };
+
+  function RenderValidity (msg, isValid) {
+    if (isValid) {
```

Comment:
Before this, it's a good idea to remove both classes to make sure we don't mix them (first it was invalid and then valid) to avoid an unwanted behavior. There's a jQuery method called `removeClass` that does that separating classes by space.

Reply:
Fixed now.  See comment below regarding 'attr'.